Add caters to PWA on SPA page (#614)
piwik/developer-documentation: docs/4.x/spa-tracking.md

category: Integrate
---

# Single-Page Application/Progressive Web App Tracking

> Single-page websites and Progressive Web App have become a standard over the last years. Getting the tracking of such websites and apps right is crucial to your success as you need to ensure the measured data is meaningful and correct.

Reviewer: ```suggestion
Single-page websites and Progressive Web Apps have become a standard over the last years. Getting the tracking of such websites and apps right is crucial to your success as you need to ensure the measured data is meaningful and correct.
```

2. or in your Single Page App, if you are using the 'Pageview Trigger' to trigger a Pageview, you can trigger a Tag Manager Event `{event: 'mtm.PageView'}` by calling the following line in JavaScript: `_mtm.push({'event': 'mtm.PageView'});`. 
   - This would also work similarly when you use instead the 'DOM Ready Trigger' (call `_mtm.push({'event': 'DOMReady'});`) or when you use the 'Window Loaded Trigger' (call `_mtm.push({'event': 'WindowLoad'});`

## Offline Tracking

> If your web application require offline tracking, refer [Matomo Offline Tracking](https://matomo.org/faq/how-to/how-do-i-set-up-matomo-offline-tracking/)

Reviewer: ```suggestion
If your web application requires offline tracking, please refer to the [Matomo  offline tracking](https://matomo.org/faq/how-to/how-do-i-set-up-matomo-offline-tracking/) guide.
```

> If your web application requires offline tracking, please refer to the [Matomo  offline tracking](https://matomo.org/faq/how-to/how-do-i-set-up-matomo-offline-tracking/) guide.

Reviewer 2: ```suggestion
If your web application requires offline tracking, please refer to the [Matomo offline tracking](https://matomo.org/faq/how-to/how-do-i-set-up-matomo-offline-tracking/) guide.
```
